### PR TITLE
Cc compare spectra

### PR DESCRIPTION
**Pull request by chuntian236**
Created "compare_utils.py" for spectra comparison and alignment: 
https://github.com/AI-multimodal/Lightshow/blob/cc-compare-spectra/lightshow/postprocess/compare_utils.py

Jupyter notebook: 
https://github.com/AI-multimodal/Lightshow/blob/cc-compare-spectra/notebooks/compare_spectra.ipynb

### CODE REVIEW

File: lightshow/postprocess/compare_utils.py [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+    start = seq.argmax()
+
+    logic = x < x[start] + erange
+    seq = x == x[logic][-1]
+    end = seq.argmax()
+
+    return start, end
+
+
+def CosSimilar(v1, v2):
```
Comment (matthewcarbone) [inline lightshow/postprocess/compare_utils.py:79]: Minor ask here, can we make the function styles consistent? I.e., this should be `cos_similar`.

File: lightshow/postprocess/compare_utils.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+                % decay
+            )
+        pearson *= decay
+        spearman *= decay
+        coss *= decay
+    correlation = [ii for ii in [pearson, spearman, coss] if not np.isnan(ii)]
+    return correlation
+
+
+def maxCorr(
```
Comment (matthewcarbone) [inline lightshow/postprocess/compare_utils.py:170]: Similar, this should be `max_corr`.

File: lightshow/postprocess/compare_utils.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+    cosine similarity : real; v1 * v2 / (norm(v1) * norm(v2))
+    """
+    norm1 = np.sqrt(np.dot(v1, v1))
+    norm2 = np.sqrt(np.dot(v2, v2))
+    cosSimilarity = np.dot(v1, v2) / (norm1 * norm2)
+    return cosSimilarity
+
+
+def spectraCorr(
+    spectrum1, spectrum2, omega=0, GRID=None, verbose=True, method="all"
```
Comment (matthewcarbone) [inline lightshow/postprocess/compare_utils.py:98]: All caps are generally reserved for globals. Can we change `GRID` to `grid`?

**Review by matthewcarbone — CHANGES_REQUESTED** [review]
Overall very nice. Just a few nitpicks then I can merge. Thanks!

**Review by matthewcarbone — CHANGES_REQUESTED** [review]
I think the code looks good but we need to improve the documentation here a bit more. Docs should follow the [NumPy style](https://numpydoc.readthedocs.io/en/latest/format.html). @chuntian236, do you want to do this here or should we just open an issue and do it later?

**Comment by chuntian236** [comment]
> I think the code looks good but we need to improve the documentation here a bit more. Docs should follow the [NumPy style](https://numpydoc.readthedocs.io/en/latest/format.html). @chuntian236, do you want to do this here or should we just open an issue and do it later?

Let me try to do this here. I might forget everything if I decide to do it later.

**Review by matthewcarbone — APPROVED** [review]
This is way better. Some nitpicks here and there but honestly I don't really mind merging at this stage.

**Pull request merged**